solidity/contracts/token/libs/MovableCollateralRouter.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.0;
+
+import {FungibleTokenRouter} from "./FungibleTokenRouter.sol";
+import {ValueTransferBridge} from "./ValueTransferBridge.sol";
+
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+import {Context} from "@openzeppelin/contracts/utils/Context.sol";
+import {ContextUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/ContextUpgradeable.sol";
+
+abstract contract MovableCollateralRouter is AccessControl {
+    constructor() {
+        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
+    bytes32 public constant REBALANCER_ROLE = keccak256("REBALANCER_ROLE");
+
+    mapping(uint32 destinationDomain => bytes32 recipient)
+        public allowedDestinations;
+    mapping(uint32 destinationDomain => mapping(ValueTransferBridge bridge => bool isValidBridge))
+        public allowedBridges;
+
+    event CollateralMoved(
+        uint32 indexed domain,
+        bytes32 recipient,
+        uint256 amount,
+        address indexed rebalancer
+    );
+
+    error BadDestination(address rebalancer, uint32 domain);
+    error BadBridge(address rebalancer, ValueTransferBridge bridge);
+
+    function moveCollateral(
```

Comment:
move collateral sounds a little scary
should we rename to rebalance?

Reply:
still hesitant to allow arbitrary recipients here, feels like a footgun

Reply:
Done!